src/rust/engine/src/nodes.rs

```diff
@@ -583,39 +583,124 @@ impl SelectDependencies {
     //   https://github.com/pantsbuild/pants/issues/4020
 
     let dep_subject_key = externs::key_for(dep_subject);
-    if self.selector.transitive {
-      // After the root has been expanded, a traversal continues with dep_product == product.
-      let mut selector = self.selector.clone();
-      selector.dep_product = selector.product;
-      context.get(
-        SelectDependencies {
-          subject: dep_subject_key,
-          variants: self.variants.clone(),
-          selector: selector,
-        }
+    context.get(Select::new(self.selector.product.clone(), dep_subject_key, self.variants.clone()))
+  }
+
+  fn store(&self, dep_product: &Value, dep_values: Vec<&Value>) -> Value {
+    // Not an inner node, or not a traversal.
```

Comment:
Comment is probably stale. But also, you can probably just inline this into the caller.